lib/std/semver.zig

```diff
@@ -0,0 +1,147 @@
+// SPDX-License-Identifier: MIT
+// Copyright (c) 2020 Zig Contributors
+// This file is part of [zig](https://ziglang.org/), which is MIT licensed.
+// The MIT license requires this copyright notice to be included in all copies
+// and substantial portions of the software.
+const std = @import("std");
+
+/// A software version formatted according to the Semantic Version 2 specification.
+///
+/// See: https://semver.org
+pub const SemanticVersion = struct {
```

Comment:
This seems like it would work well using the file as the struct.

Reply:
Interesting, are there other files in the standard library that follow this convention that I can look at? Fields become global variables and self-references become `@This()`?

Reply:
`std.mem.Allocator` uses this; the file's at https://github.com/ziglang/zig/blob/master/lib/std/mem/Allocator.zig

Reply:
I implemented this suggestion. Thanks for the tips.